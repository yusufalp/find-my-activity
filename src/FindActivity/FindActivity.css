.find-button {
  margin-right: 100%;
  width: 100%;
  margin-top: 24px;
}