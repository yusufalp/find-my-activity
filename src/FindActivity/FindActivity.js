import React from 'react';
import ActivityContext from '../ActivityContext/ActivityContext';
import './FindActivity.css';

class FindActivity extends React.Component {
  static contextType = ActivityContext;
  state = {
    results: []
  }

  handleSubmit(e) {
    e.preventDefault();
    const category = e.target.category.value
    const results = this.context.activities.filter(activity =>
      activity.category === category)
    this.setState({
      results
    })
  }

  render() {
    return (
      <div>
        <header className="find-header">
          <h2>Find Activity</h2>
          <p>
            In this section, you can search activities from the categories listed below and
            click find. The results will be listed below.
          </p>
        </header>
        <section>
          <form onSubmit={e => this.handleSubmit(e)}>
            <label htmlFor="category">Categories</label>
            <select name="category" id="category" required>
              <option value="">Select a category</option>
              {this.context.categories.map((category, i) =>
                <option
                  key={i}
                  value={category.category}
                >
                  {category.category}
                </option>
              )}
            </select>
            <button
              className="find-button"
              type="submit"
            >Find</button>
          </form>
        </section>
        <hr />
        {
          this.state.results.map((result, i) =>
            <section key={i}>
              <h2>Name: {result.name}</h2>
              <p>Details: {result.content}</p>
              <p>Duration: {result.duration}</p>
              <p>Materials: {result.materials}</p>
              <p>Age Group: {result.agegroup}</p>
            </section>
          )
        }
      </div>
    )
  }
}

export default FindActivity;