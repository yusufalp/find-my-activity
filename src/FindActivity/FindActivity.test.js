import React from 'react';
import ReactDOM from 'react-dom';
import FindActivity from './FindActivity';
import renderer from 'react-test-renderer';

describe('FindActivity component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FindActivity />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<FindActivity />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})