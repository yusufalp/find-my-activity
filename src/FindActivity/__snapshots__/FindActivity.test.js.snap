// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindActivity component renders the UI as expected 1`] = `
<div>
  <header
    className="find-header"
  >
    <h2>
      Find Activity
    </h2>
    <p>
      In this section, you can search activities from the categories listed below and click find. The results will be listed below.
    </p>
  </header>
  <section>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="category"
      >
        Categories
      </label>
      <select
        id="category"
        name="category"
        required={true}
      >
        <option
          value=""
        >
          Select a category
        </option>
      </select>
      <button
        className="find-button"
        type="submit"
      >
        Find
      </button>
    </form>
  </section>
  <hr />
  <img
    alt="kids searching"
    className="group-kids"
    src="/images/children-searching.png"
  />
</div>
`;
